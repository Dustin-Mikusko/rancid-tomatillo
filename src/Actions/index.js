export const setMovies = movies => ({
  type: 'SET_MOVIES',
  movies
});

export const changeLoading = isLoading => ({
  type: 'CHANGE_LOADING',
  isLoading: !isLoading
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});
