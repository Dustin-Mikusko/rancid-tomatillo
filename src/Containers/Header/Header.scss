header {
  border: 6px ridge white;
  position: relative;
  margin: 12px;
  margin-bottom: 75px;
}

h1 {
  font-family: 'Limelight', cursive;
  color: #FFF;
  font-size: 400%;
  padding: 10px;
  margin: 12px;
}

.title-link {
  text-decoration: none;
}

.logout-link,
.login-link {
  font-family: 'Limelight', cursive;
  top: 45px;
  right: 80px;
  color: #FFF;
  text-decoration: none;
  font-size: 135%;
  margin-right: 25px;
  position: absolute;
    &:hover {
      font-style: italic;
    }
    &:active {
      transform: scale(.97)
    }
}

.logout-link {
  margin: 0;
}

