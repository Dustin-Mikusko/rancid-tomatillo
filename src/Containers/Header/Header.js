import React from 'react';
import './Header.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { signOut, changeLoading } from '../../Actions';
import PropTypes from 'prop-types';

const Header = ({ user, signOut, changeLoading }) => {
  if (!user) {
  return (
    <header>
      <Link to='/' onClick={() => changeLoading(true)} className='title-link'><h1>RANCID TOMATILLO</h1></Link>
      <Link to='/login' className='login-link'>Sign In</Link>
    </header>
  )
  }

  return (
  <header>
    <Link to='/' onClick={() => changeLoading(true)} className='title-link'><h1>RANCID TOMATILLO</h1></Link>
    <button onClick={signOut} className='logout-link'>Sign Out</button>
  </header>
  )
  //if there is a user -> show a logout button
  //this logout button will have a redirect to homepage and update global state to null user
}

const mapDispatchToProps = dispatch => ({
  signOut: () => dispatch( signOut() ),
  changeLoading: (isLoading) => dispatch( changeLoading(isLoading))
});

export default connect(null, mapDispatchToProps)(Header);

Header.propTypes = {
  user: PropTypes.object,
  signOut: PropTypes.func,
  changeLoading: PropTypes.func
}
