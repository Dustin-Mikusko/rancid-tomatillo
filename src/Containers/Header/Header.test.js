import React from 'react';
import { shallow } from 'enzyme';
import { signOut, changeLoading } from '../../Actions';
import { Header, mapDispatchToProps } from './Header';

describe('Header', () => {

  it('should match the snapshot with no user', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with a user', () => {
    const mockUser = {
      id: 4,
      name: 'Greg',
      email: 'greg@turing.io',
      ratings: []
    }
    const wrapper = shallow(<Header user={mockUser}/>);
    expect(wrapper).toMatchSnapshot()
  });

  it('should call changeLoading on link click without a user', () => {
    const mockChangeLoading = jest.fn();
    const wrapper = shallow(<Header changeLoading={mockChangeLoading}/>);

    wrapper.find('.title-link').simulate('click');

    expect(mockChangeLoading).toHaveBeenCalledWith(true);
  });

  it('should call changeLoading on link click with a user', () => {
    const mockChangeLoading = jest.fn();
    const mockUser = {
      id: 4,
      name: 'Greg',
      email: 'greg@turing.io',
      ratings: []
    }
    const wrapper = shallow(<Header changeLoading={mockChangeLoading} user={mockUser}/>);

    wrapper.find('.title-link').simulate('click');

    expect(mockChangeLoading).toHaveBeenCalledWith(true);
  });

  it('should invoke signOut on button click', () => {
    const mockSignOut = jest.fn();
    const mockUser = {
      id: 4,
      name: 'Greg',
      email: 'greg@turing.io',
      ratings: []
    };
    const wrapper = shallow(<Header signOut={mockSignOut} user={mockUser}/>);

    wrapper.find('button').simulate('click');

    expect(mockSignOut).toHaveBeenCalled();
  });

  describe('matchDispatchToProps', () => {
    let mockDispatch;

    beforeEach( () => {
      mockDispatch = jest.fn();
    });

    it('should call dispatch with the signOut action when signOut is called', () => {
      const actionToDispatch = signOut();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.signOut();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch with the changeLoading action when changeLoading is called', () => {
      const actionToDispatch = changeLoading(true);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.changeLoading(true);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

  });

});
