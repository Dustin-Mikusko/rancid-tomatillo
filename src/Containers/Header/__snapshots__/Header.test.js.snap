// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot with all correct data passing through 1`] = `ShallowWrapper {}`;
